includes/classes/Indexable/Comment/Comment.php

```diff
@@ -0,0 +1,1053 @@
+					],
+				];
+			}
+
+			$use_filters = true;
+		}
+
+		/**
+		 * Support `type` query var
+		 */
+		if ( ! empty( $query_vars['type'] ) ) {
+			$types = (array) ( is_string( $query_vars['type'] ) ? explode( ',', $query_vars['type'] ) : $query_vars['type'] );
+			$types = array_map( 'trim', $types );
+
+			$terms_map_name = 'terms';
+
+			if ( count( $types ) < 2 ) {
+				$terms_map_name = 'term';
+				$types          = $types[0];
+			}
+
+			$filter['bool']['must'][] = [
+				$terms_map_name => [
+					'comment_type.raw' => $types,
+				],
+			];
+
+			$use_filters = true;
+		}
+
+		/**
+		 * Support `type__in` query var
+		 */
+		if ( ! empty( $query_vars['type__in'] ) ) {
+			$filter['bool']['must'][]['bool']['must'] = [
+				'terms' => [
+					'comment_type.raw' => array_values( (array) $query_vars['type__in'] ),
+				],
+			];
+
+			$use_filters = true;
+		}
+
+		/**
+		 * Support `type__not_in` query var
+		 */
+		if ( ! empty( $query_vars['type__not_in'] ) ) {
+			$filter['bool']['must'][]['bool']['must_not'] = [
+				'terms' => [
+					'comment_type.raw' => array_values( (array) $query_vars['type__not_in'] ),
+				],
+			];
+
+			$use_filters = true;
+		}
+
+		if ( $use_filters ) {
+			$formatted_args['post_filter'] = $filter;
+		}
+
+		return apply_filters( 'ep_comment_formatted_args', $formatted_args, $query_vars );
+	}
+
+	/**
+	 * Put mapping for comments
+	 *
+	 * @since  3.1
+	 * @return boolean
+	 */
+	public function put_mapping() {
+		$es_version = Elasticsearch::factory()->get_elasticsearch_version();
+
+		if ( empty( $es_version ) ) {
+			$es_version = apply_filters( 'ep_fallback_elasticsearch_version', '2.0' );
+		}
+
+		$mapping_file = 'initial.php';
+
+		if ( version_compare( $es_version, '5.0', '<' ) ) {
+			$mapping_file = 'pre-5-0.php';
+		} elseif ( version_compare( $es_version, '7.0', '>=' ) ) {
+			$mapping_file = '7-0.php';
+		}
+
+		$mapping = require apply_filters( 'ep_comment_mapping_file', __DIR__ . '/../../../mappings/comment/' . $mapping_file );
+
+		$mapping = apply_filters( 'ep_comment_mapping', $mapping );
+
+		return Elasticsearch::factory()->put_mapping( $this->get_index_name(), $mapping );
+	}
+
+	/**
+	 * Query DB for comments
+	 *
+	 * @param  array $args Query arguments
+	 * @since  3.1
+	 * @return array
+	 */
+	public function query_db( $args ) {
+		$all_query = new WP_Comment_Query(
+			[
+				'count' => true,
+			]
+		);
+
+		$defaults = [
+			'number'  => $this->get_bulk_items_per_page(),
+			'offset'  => 0,
+			'orderby' => 'comment_ID',
+			'order'   => 'desc',
+		];
+
+		if ( isset( $args['per_page'] ) ) {
+			$args['number'] = $args['per_page'];
+		}
+
+		$args = apply_filters( 'ep_comment_query_db_args', wp_parse_args( $args, $defaults ) );
+
+		$query = new WP_Comment_Query( $args );
+
+		array_walk( $query->comments, [ $this, 'remap_comments' ] );
+
+		return [
+			'objects'       => $query->comments,
+			'total_objects' => absint( $all_query ),
```

Comment:
This call should use `WP_Comment_Query->get_comments()` method, otherwise `absint` would always return 1 casting the actual class rather than the integer.

```suggestion
			'total_objects' => absint( $all_query->get_comments() ),
```